tests/phpunit/tests/dependencies/scripts.php

```diff
@@ -2996,10 +3008,31 @@ protected function parse_markup_fragment( $markup ) {
 	 * @param string $message  Message.
 	 */
 	protected function assertEqualMarkup( $expected, $actual, $message = '' ) {
+		$expected = str_replace( " type='text/javascript'", '', $expected );
+		$expected = str_replace( ' type="text/javascript"', '', $expected );
+		$expected = str_replace( "/* <![CDATA[ */\n", '', $expected );
+		$expected = str_replace( "\n/* ]]> */", '', $expected );
+		$expected = str_replace( ' defer="defer"', ' defer', $expected );
+		$expected = str_replace( ' async="async"', ' async', $expected );
+
+		$actual = str_replace( " type='text/javascript'", '', $actual );
+		$actual = str_replace( ' type="text/javascript"', '', $actual );
+		$actual = str_replace( "/* <![CDATA[ */\n", '', $actual );
+		$actual = str_replace( "\n/* ]]> */", '', $actual );
+		$actual = str_replace( ' defer="defer"', ' defer', $actual );
+		$actual = str_replace( ' async="async"', ' async', $actual );
```

Comment:
Not sure I'm a fan of having all these exceptions here. Can we somehow ensure the tests pass the relevant values instead? The `CDATA` one feels okay to me, but otherwise, I think starting to have exceptions here sets a bad precedent that could eventually move this assertion further away from asserting "equal markup".

Maybe instead create a more specific `assertEqualScriptTag` or something like that with these exceptions?

Reply:
How about 3ba5135273ae0bb21358e86b0a061aae9e84a74b? I improved the normalization process to use the DOM instead, so it will be much more robust and safe. I also made it clear from the method description that it compares with normalizations applied. Lastly, since this method is contained in the `Tests_Dependencies_Scripts` class I think this also sufficiently indicates that it is specific to checking `script` tag markup.

Reply:
The benefit of `DOMDocument` is PHPUnit supports comparing two instances in `assertEquals`, and it normalizes non-semantic differences (e.g. attribute order). See https://github.com/WordPress/wordpress-develop/pull/4773#discussion_r1333266060

Reply:
In other words, `assertEqual` is intentionally not strict in how it compares equality. It checks semantic equality as opposed to `assertSame`. So `==` instead of `===`. So I think it makes sense.